pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/source/PulsarSource.java

```diff
@@ -36,20 +53,16 @@
 import org.apache.pulsar.io.core.Source;
 import org.jboss.util.Classes;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.concurrent.TimeUnit;
-
 @Slf4j
-public class PulsarSource<T> implements Source<T> {
```

Comment:
since we are pretty much make the Pulsar source into a push source here, can we just use the following push source abstract class:

https://github.com/apache/incubator-pulsar/blob/master/pulsar-io/core/src/main/java/org/apache/pulsar/io/core/PushSource.java

Reply:
this looked completely different before I merged master...